ParkingFinder/handlers/user_information.py

```diff
@@ -41,6 +43,78 @@ def get(self, user_id):
+        # TODO : token validation
+        try:
+            assert user_id
+            user = yield UserService.get_user_detail(
+                user_id=user_id
+            )
+            current_activated_vehicle = (user.activated_vehicle if user.activated_vehicle is None else user.activated_vehicle)
+            current_vehicle_list = ([] if user.owned_vehicles is None else user.owned_vehicles)
+            passed_in_activated_vehicle = self.get_body_argument(name='activated_vehicle', default=None)
+            passed_new_vehicles = self.get_body_argument(name='new_vehicles', default=[])
+
+            # get the current user's car plate number to make checking easier
+            current_vehicle_plate_list = []
+            for exist_vehicle in current_vehicle_list:
+                current_vehicle_plate_list.append(exist_vehicle.plate)
+            passed_new_vehicles_plate_list = []
+            for new_vehicle in passed_new_vehicles:
+                passed_new_vehicles_plate_list.append(new_vehicle['plate'])
+
+            if not passed_new_vehicles:
+                for new_vehicle in passed_new_vehicles:
+                    if not (new_vehicle['plate'] in current_vehicle_plate_list):
+                        VehicleRepository.insert_registered_vehicle(user_id,
+                                                                    Vehicle({
+                                                                        'plate': new_vehicle['plate'],
+                                                                        'brand': new_vehicle['brand'],
+                                                                        'model': new_vehicle['model'],
+                                                                        'color': new_vehicle['color'],
+                                                                        'year': new_vehicle['year']
+                                                                    })
+                                                                    )
+            if not (passed_in_activated_vehicle is None):
+                if not (current_activated_vehicle is None):
+                    if not (passed_in_activated_vehicle == current_activated_vehicle):
+                        if (passed_in_activated_vehicle in current_vehicle_plate_list) \
+                                or (passed_in_activated_vehicle in passed_new_vehicles_plate_list):
+                            user.activated_vehicle = passed_in_activated_vehicle
+                            UserRepository.upsert(user)
+                            self.write(passed_in_activated_vehicle)
+                        else:
+                            self.set_status(httplib.BAD_REQUEST)
+                            self.write({
+                                'error': "please activated the car that is being registered"
+                            })
+
+            if (not passed_new_vehicles) and (passed_in_activated_vehicle is None):
```

Comment:
payload <= arguments
if payload.get('activate_vehicle'):
  service.activate_vehicle(activate_vehicle_plate, user_id)
if payload.get('register_vehicles')
  entities = [mapper to entity for vehicle in register_vehicles]
  service.register_vehicles(entities)